From 459ae818fc8dae717549810150de4d191ce158f1 Mon Sep 17 00:00:00 2001
From: micrictor <mic.ric.tor@gmail.com>
Date: Sat, 22 Feb 2020 12:17:25 -0800
Subject: [PATCH] Fix stack buffer overflow in yaffsfs_istat

Prevent a stack buffer overflow in yaffsfs_istat by increasing the buffer size to the size required by tsk_fs_time_to_str.
---
 tsk/fs/yaffs.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tsk/fs/yaffs.cpp b/tsk/fs/yaffs.cpp
index 68d8b0219b..f5ec30a09d 100755
--- a/tsk/fs/yaffs.cpp
+++ b/tsk/fs/yaffs.cpp
@@ -2439,7 +2439,7 @@ static uint8_t
     YAFFSFS_INFO *yfs = (YAFFSFS_INFO *)fs;
     char ls[12];
     YAFFSFS_PRINT_ADDR print;
-    char timeBuf[32];
+    char timeBuf[128];
     YaffsCacheObject * obj = NULL;
     YaffsCacheVersion * version = NULL;
     YaffsHeader * header = NULL;
